### DISCUSSION TITLE
Smoothing process

### DISCUSSION THREAD

**Discussion by Rahim-Habibi** [libMesh/libmesh#3222] [General]
Hi everyone,

I am simulating a large scale geomechanical model including fault. In order to avoid sharp change of the material properties near the fault, I decided to apply phase-field approach for properly modeling. But, as you probably guess, by doing so much time is needed for convergence. Then, I just found Smoothed Finite Element, which is more or less different in creating shape function in respect to standard FEM, but it has higher convergence rate.
Now, I wondering to know, is there any way to control the shape functions in LibMesh. Since, based on the formulation of the Smoothed FEM, after obtaining the displacement using standard FEM, smoothing process is being performed on elements resulting a smooth matrix. So, I want to know whether it is possible to perform smoothing process in LibMesh.

Thank you in advance

**Comment by roystgnr** [comment]
> is there any way to control the shape functions in LibMesh

libMesh is written in C++, which is Turing-complete, so technically I've got to say: "yes".  But if you're looking for an existing Smoothed Finite Element implementation, we don't have one.  You'd have to either write your own new FE type or postprocess the outputs of an existing FE type in user code.
